cmd/list/version/cmd.go

```diff
@@ -67,7 +73,23 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
-	var availableVersions []*cmv1.Version
+	var (
+		hcpVersions       []*cmv1.Version
+		availableVersions []*cmv1.Version
+	)
+
+	// Create a separate slice of only hcp-enabled versions
+	for _, version := range versions {
+		if !version.HostedControlPlaneEnabled() {
+			continue
+		}
+		hcpVersions = append(hcpVersions, version)
+	}
+
+	// If hosted-cp arg is supplied, use the hcp enabled versions
+	if args.hostedCp {
+		versions = hcpVersions
+	}
 
 	// Remove disabled versions
 	for _, version := range versions {
```

Comment:
What do you think of adding the if args.hostedCp && !version.HostedControlPlaneEnabled in the loop that sets up available versions instead of separating the slice?

Reply:
Smart, let me do that